src/main/java/duke/tasks/TaskList.java

```diff
@@ -0,0 +1,358 @@
+package duke.tasks;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.IntStream;
+
+import duke.exceptions.DukeException;
+import duke.storage.Storage;
+
+/**
+ * The TaskList class manages a list of tasks.
+ * It follows the Singleton pattern to ensure only one instance exists.
+ */
+public class TaskList {
+    private static TaskList taskList;
+    private final List<Task> taskStore;
+    private final DateTimeFormatter formatter = DateTimeFormatter
+            .ofPattern("yyyy-MM-dd HHmm");
+
+    /**
+     * Constructs a new TaskList object.
+     */
+    public TaskList() {
+        this.taskStore = new ArrayList<>();
+    }
+
+    /**
+     * Gets the singleton instance of the TaskList class.
+     *
+     * @return The TaskList instance
+     */
+    public static TaskList getInstance() {
+        if (taskList == null) {
+            taskList = new TaskList();
+        }
+        return taskList;
+    }
+
+    /**
+     * Loads a task from a string representation and adds it to the task list.
+     *
+     * @param data String representation of the task
+     */
+    public void loadData(String data) {
+        DateTimeFormatter dtf = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm");
+
+        Arrays.stream(data.split("\n"))
+                .map(line -> line.split("\\s*\\|\\s*"))
+                .forEach(parts -> {
+                    try {
+                        Task task = switch (parts[0]) {
```

Comment:
I like the use of stream and splitting the string to parse out the type of command